packages/workflow-compile/index.js

```diff
@@ -59,7 +61,7 @@ async function compile(config) {
 }
 
 const WorkflowCompile = {
-  async compile(options) {
+  async compile(options, loadDb = false) {
```

Comment:
We should avoid adding any arguments to this function - it should be based only on `config.db.enabled`

Reply:
I added this because we wanted to distinguish between actual compilation and compilation from test. So I pass the true flag from the compile command. Was there a better way to do that? `config.db.enabled` only tells us that the user has enabled using the db, but doesn't tell us whether they're running `truffle test`.

Reply:
I reworked this but don't love it. There's config.loadDb that I added in the compile command file, and then there's { names: true} being passed to `loadCompilations`. I added the names flag in the name logic PR, but looking at it now I feel like couldn't we drop it? Because if we are running `truffle test` we won't get to `loadCompilations` anyway. What do you think?